### ISSUE TITLE
Initial axis labels have tiny font; fixes itself with slight resize

### ISSUE THREAD

**Issue by sammyjava** [neuhausi/canvasXpress#33]
I'm stumped on how to get my heatmap to initialize with the axis label font size that I want. I've attached the before and after: before a slight resize and after. The resize "restores" the fonts to the desired size, but the initial heatmap always shows the labels, and the heat map index, in a very tiny font/size. I still haven't figured out how to get the heatmap index to enlarge; it doesn't change with a resize, not that that's a solution, anyway.
![heatmap-before-after-resizing](https://cloud.githubusercontent.com/assets/5657219/26068947/932f8ea8-395c-11e7-930d-d836583b4b89.png)

**Comment by sammyjava** [comment]
Here's what I'm sending:
```
var cx = new CanvasXpress("canvasx",
    {"y":{"data":[[29.5],[4.24],[53.56],[27.91],[30.46],[28.03],[21.1],[26.27],[29.27],[28.4],[83.34]],"smps":["ICC4958.Ca_03061"],"vars":["GS","YL","SAM","FB1","FB2","FB3","FB4","F1","F2","F3","F4"]}},
    {
                               "graphType": "Heatmap",
                               "title": "RNA-seq expression (TPM)",
                               "titleScaleFontFactor": 0.3,
                               "xAxisTitle": "",
                               "fontName": "Arial",
                               "canvasBox": true,
                               "heatmapCellBox": true,
                               "smpLabelScaleFontFactor": 10.0,
                               "smpLabelFontStyle": "plain",
                               "smpLabelFontColor": "black",
                               "varLabelScaleFontFactor": 10.0,
                               "varLabelFontStyle": "plain",
                               "varLabelFontColor": "black",
                               "varLabelRotate": 0,
                               "colorSpectrum": [ "white","yellow","orange","gold","red" ],
                               "colorSpectrumBreaks": [ 1,10,100,1000,10000 ],
                               "isLogData": false,
                               "heatmapCellBox": true,
                               "showHeatMapIndicator": true,
                               "heatmapIndicatorPosition": "top",
                               "heatmapIndicatorHistogram": false,
                               "heatmapIndicatorWidth": 200,
                               "heatmapIndicatorHeight": 10,
                               "samplesKmeaned": false,
                               "variablesKmeaned": false,
                               "samplesClustered": false,
                               "variablesClustered": false,
                               "showSmpDendrogram": false,
                               "showVarDendrogram": false
                           }
 );
```

**Comment by neuhausi** [comment]
You are adding "smpLabelFontStyle": "plain" and "varLabelFontStyle": "plain" which are not recognized. If you want the regular style then don't include them in the parameters or just specify them as "". Also the ScaleFontFactors are quite high. Just reduce them to ~1.5. There is a bug in showing the sample labels and I will fixe that in Version 16.6.
Sorry about the delay

**Comment by sammyjava** [comment]
Thanks! No prob on the delay. How about the heatmap index? It's tiiiiiny.

**Comment by neuhausi** [comment]
You can adjust the indicator dimensions the following way:

`"heatmapIndicatorWidth": 2000, // the width of the indicator`
`"heatmapIndicatorHeight": 100, // the height of the indicator`
`"legendScaleFontFactor": 10, // the font size for the indicator`

I still need to fix one bug in the sample labels.

**Comment by neuhausi** [comment]
I had to add a new parameter to prevent scaling the heatmap indicator because setting automatically all the font sizes and dimension of the graph can be quite complicated. The new parameter is called adjustAspectRatioHeatmapIndicator and it needs to be set to false (defult is true) to honor the dimensions of the heatmapIndicatorWidth and heatmapIndicatorHeight in pixels. The only thing that would need user/developer adjustment will the the scaling factors for the fonts. Here is the parameters in the new version that I have not published yet:
```
"graphType": "Heatmap",
"colorSpectrum": [ "white","yellow","orange","gold","red" ],
"colorSpectrumBreaks": [ 1,10,100,1000,10000 ],
"fontName": "Arial",
"canvasBox": true,
"title": "RNA-seq expression (TPM)",
"titleScaleFontFactor": 0.5,
"adjustAspectRatio": true,
"adjustAspectRatioHeatmapIndicator": false,
"heatmapCellBox": true,
"heatmapIndicatorPosition": "top",
"heatmapIndicatorHistogram": false,
"legendScaleFontFactor": 5,
"smpLabelFontColor": "black",
"smpLabelScaleFontFactor": 4,
"varLabelFontColor": "black",
"varLabelScaleFontFactor": 1.5
```
And here is the graph:
![cx-canvas](https://user-images.githubusercontent.com/254923/28149823-654ac37e-675e-11e7-9bc2-9c5e79b2b897.png)

The ideal case for heatmaps is to plot more than five samples and five variables. Personally I wouldn't use a heatmap here but rather a bar graph.

I will publish the new version before the end of the week

Isaac

**Comment by sammyjava** [comment]
Cool, thanks! Yeah there's limited vertical space for a bar graph and folks are used to seeing heat maps for expression on a different part of the site, so this is a slice of that other normal heat map. Appreciate the update!

**Comment by neuhausi** [comment]
Fixed in version 16.6
